libpod/networking.go

```diff
@@ -102,3 +108,63 @@ func (r *Runtime) teardownNetNS(ctr *Container) error {
 
 	return nil
 }
+
+//ReadPortRangeFile reads the range of available ports per proc
+func ReadPortRangeFile() (int, int, error) {
+	portFileName := "/proc/sys/net/ipv4/ip_local_port_range"
+	portFile, err := ioutil.ReadFile(portFileName)
+	if err != nil {
+		return 0, 0, errors.Wrapf(err, "unable to read %s", portFileName)
+	}
+	portFields := bytes.Fields(portFile)
+	if len(portFields) < 2 {
+		return 0, 0, fmt.Errorf("invalid data in %s", portFileName)
+	}
+	begin, err := strconv.Atoi(string(portFields[0]))
+	if err != nil {
+		return 0, 0, errors.Wrapf(err, "unable to convert port range data to int")
+	}
+
+	end, err := strconv.Atoi(string(portFields[1]))
+	if err != nil {
+		return 0, 0, errors.Wrapf(err, "unable to convert port range data to int")
+	}
+
+	return begin, end, nil
+}
+
+//GetRandomPort returns a unused port given the port range as defined in /proc
+func GetRandomPort() (int, error) {
+	start, end, err := ReadPortRangeFile()
+	if err != nil {
+		return -1, err
+	}
+	portRange := end - start
+	for i := 0; i < portRange; i++ {
+		// Generates a number between 0 and the delta of the port range
+		randomNum := rand.Intn(portRange)
+		// Add randomNum to the start to get random port
+		randomPort := start + randomNum
+		// Now make sure port is not used
+		returnCode, err := utils.ExecCmdReturnCode("fuser", fmt.Sprintf("%d/tcp", randomPort))
```

Comment:
Any way we can do this internally rather then executing a separate command.

Reply:
Also adds a requirement on psmisc for podman.

Reply:
fuser is shoiuld be installed everwhere ... its on atomic systems as well iirc ... im open to ideas if you have them. my udnerstanding is that this is in /proc but you need the pid to find it.

Reply:
fixed